### DISCUSSION TITLE
How do I pass DefineConstants to my reference project?

### DISCUSSION THREAD

**Discussion by wf-soft** [nuke-build/nuke#999] [Q&A] [answered]
I use
```Csharp
.AddProperty ("DefineConstants", "TEST")
```
 but it only works on the main project and has no effect on referenced dependent projects

**Comment by ron-myers** [comment, marked as answer]
Could you expand on your use case @wf-soft ?  `AddProperty` is wildly available on tools.

**Reply by wf-soft** [reply]
@ron-myers I have a main project and a reference project, and I use the following code to generate conditional compilation, but only the main project is affected by conditional compilation and has no effect on the reference project

```Csharp
DotNetPublish(s => s
                .SetProject(project)
                .SetConfiguration(Configuration)
                .SetPublishProfile("FolderProfile")
                .SetOutput(OutputDirectory)
                .AddProperty("DefineConstants", "USER")
                .SetWarningLevel(0)
```

**Reply by ron-myers** [reply]
I believe you need to add the property when you compile - not just publish @wf-soft

**Reply by wf-soft** [reply]
@ron-myers How should I add compile time properties for a reference project from nuke

**Reply by ron-myers** [reply]
@wf-soft  When you call `DotNetBuild`~~`DotNetCompile`~~ add `.AddProperty("DefineConstants", "USER")` to that call.

**Reply by wf-soft** [reply]
@ron-myers Nuke doesn't seem to have such an DotNetCompile

**Reply by ron-myers** [reply]
Apologies - I think of it as compile, but the call is `DotNetBuild`.

**Reply by wf-soft** [reply]
@ron-myers I found that because I didn't add a definition into Assemblyinfo.cs of the WPF reference project that I couldn't see the change, the Nuke conditional compilation actually worked, Thank you for your answer
